docs/source/sample_datasets.rst

```diff
@@ -34,24 +34,42 @@ has no physical significance.
 
 **Raw file:**
 
-.. testcode:: createraw
+.. testsetup:: sampledataraw
+
+    import os
+    os.mkdir("temp_sample_raw")
+    os.chdir("temp_sample_raw")
```

Comment:
I would prefer if the `TemporaryDirectory` function were used here: see https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

This will automatically create the new directory in the temporary directory of the operating system, provides a `cleanup()` method etc.